lightning/src/ln/channel.rs

```diff
@@ -2512,7 +2512,12 @@ impl<Signer: WriteableEcdsaChannelSigner> Channel<Signer> {
 		                                          obscure_factor,
 		                                          holder_commitment_tx, best_block, self.context.counterparty_node_id);
 
-		channel_monitor.provide_latest_counterparty_commitment_tx(counterparty_initial_bitcoin_tx.txid, Vec::new(), self.context.cur_counterparty_commitment_transaction_number, self.context.counterparty_cur_commitment_point.unwrap(), logger);
+		channel_monitor.provide_initial_counterparty_commitment_tx(
```

Comment:
Hm, storing the whole commitment tx object is 488 bytes whereas the current is 52 bytes. Maybe in the grand scheme of a monitor it might not be thaat much, but I feel like that's quite a bit more to store just to save on a computation we only expect the user to call once?

Reply:
Oh I was just going off the `CommitmentTransaction` struct - if we stored the built `Transaction` it's less, but then we'd still probably want to rebuild the `CommitmentTransaction` struct when we provide it to the user?